src/OpenTK/MouseCursor.cs

```diff
@@ -28,28 +28,55 @@
 #endregion
 
 using System;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace OpenTK
 {
     /// <summary>
+    /// Stores information about frame timing
+    /// </summary>
+    public struct MouseCursorFrameInformation
+    {
+        /// <summary>
+        /// The frames time in seconds
+        /// </summary>
+        public float duration;
+
+        /// <summary>
+        /// The associated frame
+        /// </summary>
+        public MouseCursorFrame frame;
+    }
+
+    /// <summary>
     /// Represents a predefined or custom mouse cursor.
+    /// new MouseCursor() represents an empty animation. Use MouseCursor.Default instead.
     /// </summary>
-    public sealed class MouseCursor : WindowIcon
+    public sealed class MouseCursor : IEnumerable<MouseCursorFrameInformation>
```

Comment:
While normally a good idea and best practice, implementing `IEnumerable<T>` isn't the best idea here:
 - It's going to cause virtual method calls
 - It's going to create memory pressure from enumerable GC.
 - It prevents O(1) access to elements.

You could look at an alternative like IReadOnlyList<T>, but we'd still be providing an API that allows for two of the above problems, of which GC is the most significant.

My suggestion then is just to provide a read-only indexer property and a count method. Simple, no GC, and fast.